feature/interface/staticarp/otg_tests/static_arp_test/static_arp_test.go

```diff
@@ -0,0 +1,415 @@
+// and ate:port2 as the destination interface.  The traffic should
+// flow as expected both when using dynamic or static ARP since the
+// Ixia interfaces are promiscuous.  However, using custom egress
+// filter, we can tell if the static ARP is honored or not.
+//
+// Synthesized static MAC addresses have the form 02:1a:WW:XX:YY:ZZ
+// where WW:XX:YY:ZZ are the four octets of the IPv4 in hex.  The 0x02
+// means the MAC address is locally administered.
+const (
+	plen4 = 30
+	plen6 = 126
+
+	poisonedMAC = "12:34:56:78:7a:69" // 0x7a69 = 31337
+	noStaticMAC = ""
+)
+
+var (
+	ateSrc = attrs.Attributes{
+		Name:    "port1",
+		MAC:     "00:11:01:00:00:01",
+		IPv4:    "192.0.2.1",
+		IPv6:    "2001:db8::1",
+		IPv4Len: plen4,
+		IPv6Len: plen6,
+	}
+
+	dutSrc = attrs.Attributes{
+		Desc:    "DUT to ATE source",
+		IPv4:    "192.0.2.2",
+		IPv6:    "2001:db8::2",
+		MAC:     "02:1a:c0:00:02:02", // 02:1a+192.0.2.2
+		IPv4Len: plen4,
+		IPv6Len: plen6,
+	}
+
+	dutDst = attrs.Attributes{
+		Desc:    "DUT to ATE destination",
+		IPv4:    "192.0.2.5",
+		IPv6:    "2001:db8::5",
+		MAC:     "02:1a:c0:00:02:05", // 02:1a+192.0.2.5
+		IPv4Len: plen4,
+		IPv6Len: plen6,
+	}
+
+	ateDst = attrs.Attributes{
+		Name:    "port2",
+		MAC:     "00:12:01:00:00:01",
+		IPv4:    "192.0.2.6",
+		IPv6:    "2001:db8::6",
+		IPv4Len: plen4,
+		IPv6Len: plen6,
+	}
+)
+
+// configInterfaceDUT configures the interface on "me" with static ARP
+// of peer.  Note that peermac is used for static ARP, and not
+// peer.MAC.
+func configInterfaceDUT(i *telemetry.Interface, me, peer *attrs.Attributes, peermac string) *telemetry.Interface {
+	i.Description = ygot.String(me.Desc)
+	i.Type = telemetry.IETFInterfaces_InterfaceType_ethernetCsmacd
+	if *deviations.InterfaceEnabled {
+		i.Enabled = ygot.Bool(true)
+	}
+
+	if me.MAC != "" {
+		e := i.GetOrCreateEthernet()
+		e.MacAddress = ygot.String(me.MAC)
+	}
+
+	s := i.GetOrCreateSubinterface(0)
+	s4 := s.GetOrCreateIpv4()
+	if *deviations.InterfaceEnabled {
+		s4.Enabled = ygot.Bool(true)
+	}
+	a := s4.GetOrCreateAddress(me.IPv4)
+	a.PrefixLength = ygot.Uint8(plen4)
+
+	if peermac != noStaticMAC {
+		n4 := s4.GetOrCreateNeighbor(peer.IPv4)
+		n4.LinkLayerAddress = ygot.String(peermac)
+	}
+
+	s6 := s.GetOrCreateIpv6()
+	if *deviations.InterfaceEnabled {
+		s6.Enabled = ygot.Bool(true)
+	}
+	s6.GetOrCreateAddress(me.IPv6).PrefixLength = ygot.Uint8(plen6)
+
+	if peermac != noStaticMAC {
+		n6 := s6.GetOrCreateNeighbor(peer.IPv6)
+		n6.LinkLayerAddress = ygot.String(peermac)
+	}
+
+	return i
+}
+
+func configureDUT(t *testing.T, peermac string) (interfaceOrder bool) {
+	interfaceOrder = true
+	dut := ondatra.DUT(t, "dut")
+	d := dut.Config()
+	p1 := dut.Port(t, "port1")
+	p2 := dut.Port(t, "port2")
+	i1 := &telemetry.Interface{Name: ygot.String(p1.Name())}
+	i2 := &telemetry.Interface{Name: ygot.String(p2.Name())}
+	portList := []string{*i1.Name, *i2.Name}
+	if sort.StringsAreSorted(portList) {
```

Comment:
Need to look at this test and remove the sorting

Reply:
Removed the sorting and now instead of specifying the names of dut interfaces I'm using the ondatra handle "port1" and "port2"
e.g. dut.Port(t, "port2").Name()